Fix ucj put exception when removing user

lchanmann opened the pull request:
## Ref
- https://nanopay.atlassian.net/browse/NP-7979

## Changes
- Put to bareUserCapabilityJunctionDAO to prevent rules on userCapabilityJunctionDAO from firing which could invoke sudo-ing as the deleted user then fails
- Check user in the current subject and the given subject before sudo-ing

## CI
- https://ci-jenkins.nanopay.net/view/CI/job/ci-NP-7979-Fix_tests/5/console

lchanmann committed:
Put to bareUserCapabilityJunctionDAO to prevent rules on
     userCapabilityJunctionDAO from firing which could invoke sudo-ing as the deleted user then fails.

lchanmann committed:
Fix get junction

smirnovaka commented:
are tests broken?

is the point of getJunction method to get a ucj for a particular user? i'k not sure why it would silently default to current logged in user if it was requested for somebody else(same with updating the ucj, not just reading).

lchanmann commented:
All tests pass https://ci-jenkins.nanopay.net/view/CI/job/ci-NP-7979-Fix_tests/2/console. 

The getJunctionForSubject check if you can update user context before sudo-ing otherwise it uses your context to get the ucj.

lchanmann committed:
Extract getJunctionForSubject_ to support whether to act as the subject or not

jlhughes reviewed (commented):
I'm good with this.

lchanmann committed:
Act as subject if user is not in the current subject

lchanmann committed:
Move logic for checking same user before sudo-ing to the original method as it is the single entry point to get the ucj

Can split to getJunctionForSubject_ if we need more than one entry points in the furture
